core/src/main/kotlin/io/github/paulgriffith/kindling/utils/ColumnList.kt

```diff
@@ -19,31 +18,32 @@ abstract class ColumnList<R> private constructor(
     protected inline fun <reified T> column(
         name: String? = null,
         noinline column: (TableColumnExt.(model: TableModel) -> Unit)? = null,
-        noinline value: (R) -> T
+        noinline value: (R) -> T,
     ): PropertyDelegateProvider<ColumnList<R>, ReadOnlyProperty<ColumnList<R>, Column<R, T>>> {
         return PropertyDelegateProvider { thisRef, prop ->
             val actual = Column(
                 header = name ?: prop.name,
                 getValue = value,
                 columnCustomization = column,
-                clazz = T::class.java
+                clazz = T::class.java,
             )
-            thisRef.list.add(actual)
+            thisRef.add(actual)
             ReadOnlyProperty { _, _ -> actual }
         }
     }
 
+    fun add(column: Column<R, *>) {
+        list.add(column)
+    }
+
     operator fun get(column: Column<R, *>): Int = indexOf(column)
-}
 
-fun JXTable.installColumnFactory(columns: ColumnList<*>, createColumns: Boolean = true) {
-    columnFactory = object : ColumnFactory() {
+    fun toColumnFactory() = object : ColumnFactory() {
```

Comment:
Instead of the 'createColumns' boolean, this seemed easier - gives you more flexibility when using a ColumnList